qiskit/pulse/library/symbolic_pulses.py

```diff
@@ -0,0 +1,857 @@
+            \\Bigl(-\\frac{x - \\text{duration}/2}{\\text{sigma}^2}\\Bigr)  \\Bigr),
+            \\quad 0 \\le x < \\text{duration}
+
+    where :math:`g(x)` is a standard unlifted Gaussian waveform and
+    :math:`g'(x)` is the lifted :class:`~qiskit.pulse.library.Gaussian` waveform.
+
+    References:
+        1. |citation1|_
+
+        .. _citation1: https://link.aps.org/doi/10.1103/PhysRevA.83.012308
+
+        .. |citation1| replace:: *Gambetta, J. M., Motzoi, F., Merkel, S. T. & Wilhelm, F. K.
+           Analytic control methods for high-fidelity unitary operations
+           in a weakly nonlinear oscillator. Phys. Rev. A 83, 012308 (2011).*
+
+        2. |citation2|_
+
+        .. _citation2: https://link.aps.org/doi/10.1103/PhysRevLett.103.110501
+
+        .. |citation2| replace:: *F. Motzoi, J. M. Gambetta, P. Rebentrost, and F. K. Wilhelm
+           Phys. Rev. Lett. 103, 110501 – Published 8 September 2009.*
+    """
+
+    PARAM_DEF = ["duration", "amp", "sigma", "beta"]
+
+    def __init__(
+        self,
+        duration: Union[int, ParameterExpression],
+        amp: Union[complex, ParameterExpression],
+        sigma: Union[float, ParameterExpression],
+        beta: Union[float, ParameterExpression],
+        name: Optional[str] = None,
+        limit_amplitude: Optional[bool] = None,
+    ):
+        """Initialize the drag pulse.
+
+        Args:
+            duration: Pulse length in terms of the sampling period `dt`.
+            amp: The amplitude of the Drag envelope.
+            sigma: A measure of how wide or narrow the Gaussian peak is; described mathematically
+                   in the class docstring.
+            beta: The correction amplitude.
+            name: Display name for this pulse envelope.
+            limit_amplitude: If ``True``, then limit the amplitude of the
+                waveform to 1. The default is ``True`` and the amplitude is constrained to 1.
+        """
+        if not isinstance(amp, ParameterExpression):
+            amp = complex(amp)
+
+        super().__init__(
+            duration=duration,
+            parameters=(amp, sigma, beta),
+            name=name,
+            limit_amplitude=limit_amplitude,
+        )
+
+    @classmethod
+    def _define_envelope(cls) -> "Expr":
+        import sympy as sym
+
+        t, duration, amp, sigma, beta = sym.symbols("t, duration, amp, sigma, beta")
+        center = duration / 2
+
+        gauss = amp * _lifted_gaussian(t, center, duration + 2, sigma)
+        deriv = -(t - center) / (sigma**2) * gauss
+
+        return gauss + 1j * beta * deriv
+
+    @classmethod
+    def _define_constraints(cls) -> List[Union["Expr", List]]:
+        import sympy as sym
+
+        t, duration, amp, sigma, beta, lim_amp = sym.symbols("t, duration, amp, sigma, beta, limit")
+
+        drag_eq = cls._define_envelope()
+        t_drag_max = duration / 2 - (sigma / beta) * sym.sqrt(beta**2 - sigma**2)
```

Comment:
Thanks Will this is awesome investigation. Using un-lifted version for validation makes sense (to be more conservative). However, I prefer using `sigma < abs(beta)` as a constraint, i.e. second option. With this simplification we can drop custom ITE functions (in principle they are not necessary after sympy ITE issue is solved, however actually we cannot drop them just for loading old binary). The second option seems to free us from the burdens of these temporary functions.

However, I have concern about employing option2. In this PR https://github.com/Qiskit/qiskit-terra/pull/7925 I've fixed an obvious bug in validation of drag parameter. As a consequence, it failed to load some fake backend since there was a backend-calibrated DRAG pulse with large beta. Likely validation is not correctly done by production systems, and this makes me think enabling option 2 could cause program for loading current backend (i.e. program crashes during deserializing backend cmd_def).